cmd/analyze/main.go

```diff
@@ -99,24 +99,13 @@ func main() {
 
 	sb := sandbox.New(manager.Image(), sbOpts...)
 	defer sb.Clean()
-	results := make(map[string]*dynamicanalysis.Result)
-	for _, phase := range manager.DynamicPhases() {
-		result, err := dynamicanalysis.Run(sb, pkg.Command(phase))
-		if err != nil {
-			log.Fatal("Analysis phase failed",
-				log.Label("phase", phase),
-				"error", err)
-		}
-		results[phase] = result
-		// Abort if install failed. No need to continue.
-		if result.Status != dynamicanalysis.StatusCompleted {
-			log.Error("Analysis phase failed. Cannot continue.",
-				log.Label("phase", phase),
-				log.Label("ecosystem", *ecosystem),
-				log.Label("name", *pkgName),
-				log.Label("version", *version))
-			break
-		}
+
+	results, lastRunPhase, err := analysis.RunDynamicAnalysis(sb, pkg)
+	if err != nil {
+		analysis.LogDynamicAnalysisError(pkg, lastRunPhase, err)
+		log.Fatal("Aborting due to run error", "error", err)
+	} else {
+		analysis.LogDynamicAnalysisResult(pkg, lastRunPhase, results[lastRunPhase].Status)
```

Comment:
On review, I don't given that the log messages are inside analysis, I think this logic could just be moved into the analysis package.

This would avoid accidentally breaking something when modifying it in the worker / analyze binary.

Reply:
Will do in another PR